numpy/core/tests/test_dtype.py

```diff
@@ -446,6 +447,154 @@ def test_alignment(self):
         assert_equal(t1.alignment, t2.alignment)
 
 
+def iter_struct_object_dtypes():
+    """
+    Iterates over a few complex dtypes and object pattern which
+    fill the array with a given object (defaults to a singleton).
+
+    Yields
+    ------
+    dtype : dtype
+    pattern : tuple
+        Structured tuple for use with `np.array`.
+    count : int
+        Number of objects stored in the dtype.
+    singleton : object
+        A singleton object. The returned pattern is constructed so that
+        all objects inside the datatype are set to the singleton.
+    """
+    obj = object()
+
+    dt = np.dtype([('b', 'O', (2, 3))])
+    p = ([[obj] * 3] * 2,)
+    yield dt, p, 6, obj, "<subarray>"
+
+    dt = np.dtype([('a', 'i4'), ('b', 'O', (2, 3))])
+    p = (0, [[obj] * 3] * 2)
+    yield dt, p, 6, obj, "<subarray in field>"
+
+    dt = np.dtype([('a', 'i4'),
+                   ('b', [('ba', 'O'), ('bb', 'i1')], (2, 3))])
+    p = (0, [[(obj, 0)] * 3] * 2)
+    yield dt, p, 6, obj, "<structured subarray 1>"
+
+    dt = np.dtype([('a', 'i4'),
+                   ('b', [('ba', 'O'), ('bb', 'O')], (2, 3))])
+    p = (0, [[(obj, obj)] * 3] * 2)
+    yield dt, p, 12, obj, "<structured subarray 2>"
+
```

Comment:
Turns out the better way to write this is `pytest.param(dt, p, 12, obj, id="<structured subarray 2>")`, which eliminate the need for the label` arguments below.

Reply:
oh. The pytest doc could really use a bit more example on the parameterize ref. Or maybe I just missed it.

Reply:
No, the docs are very hard to find here.